Moq.AutoMock/Resolvers/MockResolver.cs

```diff
@@ -37,7 +37,30 @@ public void Resolve(MockResolutionContext context)
             if (!(context.Value is null)) return;
 
             var mockType = typeof(Mock<>).MakeGenericType(context.RequestType);
-            if (Activator.CreateInstance(mockType, _mockBehavior) is Mock mock)
+
+            bool mayHaveDependencies = context.RequestType.IsClass 
+                                       && !typeof(Delegate).IsAssignableFrom(context.RequestType);
+            
+            object?[] constructorArgs;
+            if (mayHaveDependencies)
+            {
+                constructorArgs = context.AutoMocker.CreateArguments(context.RequestType);
+            }
+            else
+            {
+#pragma warning disable CA1825
```

Comment:
Created https://github.com/moq/Moq.AutoMocker/pull/87 as a proposed cleanup instead of disabling this here.